app/controllers/step_by_step_pages_controller.rb

```diff
@@ -78,7 +78,11 @@ def publish
 
   def schedule
     set_current_page_as_step_by_step
-    if request.post?
+  end
+
+  def schedule_datetime
+    set_current_page_as_step_by_step
+    if request.post? && params[:schedule]
```

Comment:
`schedule_datetime` is only accessible over POST, so this can be simplified to `params[:schedule]`.